static/js/wicd.js

```diff
@@ -106,14 +106,37 @@ function scan_networks() {
   $.getJSON( "scan", refresh_networks );
 }
 
-function connect_to_network() {
-  var network_id = $(this).data('id');
-  showAlert('Connecting...', 'info');
-  $.getJSON( "connect/" + network_id, function(data) {
-    showAlert('Please wait (' + data.data + ')', 'info');
-    showAlert('Connecting to ' + data.data, 'info');
-    watch_connection_status();
-  } );
+function network_status_precheck(){
+  $.getJSON("current", connect_to_network($(this).data('id')));
+}
+
+function connect_to_network(network_id) {
+  return function(current) {
+    console.log("Preparing to connect to network...", network_id, current);
+    if(current.data.network === undefined) {
+      showAlert('Connecting...', 'info');
+      $.getJSON("connect/" + network_id, function(data) {
+        showAlert('Please wait (' + data.data + ')', 'info');
+        showAlert('Connecting to ' + data.data, 'info');
```

Comment:
Wait so you're updating the alert twice in a row? Won't that first one just burn CPU cycles?